packages/flutter_web_plugins/lib/src/navigation/utils.dart

```diff
@@ -15,30 +13,30 @@ AnchorElement _urlParsingNode;
 String extractPathname(String url) {
   // TODO(mdebbar): Use the `URI` class instead?
   _urlParsingNode ??= AnchorElement();
-  _urlParsingNode.href = url;
-  final String pathname = _urlParsingNode.pathname;
+  _urlParsingNode!.href = url;
+  final String pathname = _urlParsingNode!.pathname ?? '';
```

Comment:
Making `_urlParsingNode` `final` non-null will remove the need for `!`s

Reply:
I had to eagerly initialize `_urlParsingNode` for this to work, but it's now final and non-null.